src/App.js

```diff
@@ -2,31 +2,51 @@ import React from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = React.useState(
-    JSON.parse(localStorage.getItem('savedTodoList'))|| []
-  );
+function App() {
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  const [todoList, setTodoList] = React.useState([]);
+
   React.useEffect(() => {
-    const todoListString = JSON.stringify(todoList);
-    localStorage.setItem("savedTodoList", todoListString);
-  }, [todoList]);
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
+          },
+        });
+      }, 2000);
+    }).then((result) => {
+      setIsLoading(false);
+      setTodoList(result.data.todoList);
+    });
+  }, []);
 
-  return [todoList, setTodoList];
-}
-function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  React.useEffect(() => {
+    if (isLoading === false) {
+      const todoListString = JSON.stringify(todoList);
+      localStorage.setItem("savedTodoList", todoListString);
+    }
+  }, [isLoading, todoList]);
+
+  function handleremoveTodo(id) {
+    const removeTodo = todoList.filter((todo) => id !== todo.id);
+
+    setTodoList(removeTodo);
+  }
 
   function addTodo(newTodo) {
     setTodoList([...todoList, newTodo]);
   }
 
   return (
     <>
+    <p>{isLoading ? "Loading...": "" }</p>
```

Comment:
I think we need to change this. The ternary should be between rendering `Loading...` and rendering the TodoList. How do we do that? Is this placed in the correct spot?

Reply:
Something like this?

Reply:
<p>{isLoading ? "Loading..." : <h1>Todo List</h1>}</p>

Reply:
okay I think I figured it out https://github.com/BThomas22tech/react-todo/pull/8/commits/f495e77901875cafc1396138e739cb81c5aabf8a